File: app/schemas/venue.py
from pydantic import BaseModel, EmailStr
from typing import Optional, List
from datetime import datetime

class VenueBase(BaseModel):
    venue_name: str
    address: Optional[str] = None
    contact_name: Optional[str] = None
    contact_email: Optional[EmailStr] = None
    contact_phone: Optional[str] = None
    description: Optional[str] = None
    size_sqm: Optional[float] = None
    price: Optional[float] = None
    maximum_capacity: Optional[int] = None
    currency: Optional[str] = "GBP"             # 新增字段，默认英镑
    facilities: Optional[List] = []               # 新增字段，存放设施信息（列表）
    availability: Optional[List] = []             # 新增字段，存放可用性信息（列表）
    user_has_saved :Optional[bool]= False  # <-- 新增字段
    status: Optional[str] = "ACTIVE"              # 新增字段，默认状态
    user_has_saved: Optional[bool] = False        # 新增字段，用户是否收藏

class VenueCreate(VenueBase):
    pass

class VenueUpdate(VenueBase):
    pass

class VenueInDB(VenueBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

File: frontend/src/types/venue.ts
export interface Venue {
    id: string;
    venue_name: string;
    description: string;
    address: string;
    size_sqm: number;
    price: number;
    currency: string;
    maximum_capacity: number;
    facilities: Facility[];
    availability: VenueAvailability[];
    contact_name: string;
    contact_email: string;
    contact_phone: string;
    user_has_saved: boolean;
    status: VenueStatus;
    created_at: string;
    updated_at: string;
}

export interface VenueAvailability {
    start_date: string;
    end_date: string;
    start_time: string;
    end_time: string;
}

export interface VenueBooking {
    id: string;
    venue: Venue;
    date_times: VenueAvailability[];
    total_cost: number;
    status: BookingStatus;
    created_at: string;
    updated_at: string;
}

export interface Facility {
    display_name: string;
    facility_information: string[];
}

export enum VenueStatus {
    ACTIVE = 'ACTIVE',
    INACTIVE = 'INACTIVE',
    PENDING = 'PENDING'
}

export enum BookingStatus {
    PENDING = 'PENDING',
    CONFIRMED = 'CONFIRMED',
    CANCELLED = 'CANCELLED',
    COMPLETED = 'COMPLETED'
}

export interface VenueAvailability {
    start_date: string;
    end_date: string;
    start_time: string;
    end_time: string;
}